subprojects/diagnostics/src/main/java/org/gradle/api/plugins/internal/DefaultProjectReportsPluginConvention.java

```diff
@@ -41,21 +42,24 @@ public DefaultProjectReportsPluginConvention(Project project) {
         return typeOf(ProjectReportsPluginConvention.class);
     }
 
+    @Nonnull
```

Comment:
So it seems counterintuitive to mark individual things as `@Nonnull` if that's something we generally strive for. A lot of other places use `@NonNullApi` in their `package-info.java` to represent this (and mark exceptions with `@Nullable`. Do you think that makes sense for this module as well?

Reply:
So here we have types that extend types in a package marked `@NonNullApi` and thus IJ asks you to confirm the override / implementation is `@Nonnull`.
In that case, I did not even attempt saying that this whole internal package us `@NonNullApi` tbh.

Reply:
I replaced some of these changes by the use of `@NonNullApi` and `@Nullable` where needed.
Please have another look.